### DISCUSSION TITLE
Simulation

### DISCUSSION THREAD

**Discussion by AlainZuur** [JenniNiku/gllvm#136] [General]
Hello,

I noticed that the plot() function applied to a gllvm object presents various model validation tools. Are those residuals the same as DHARMa is using (scaled quantile residuals)?

If I want to put more detail into the model validation, and utlilise other functions from DHARMa (e.g, testZeroInflation, then I assume that I can simulate 1000 data sets for *each* species, and then for each species use this:

DHARMaRes = createDHARMa(simulatedResponse = sim, observedResponse = testData$observedResponse, 
             fittedPredictedResponse = predict(fittedModel), integerResponse = T)

That would allow me to do a detailed model validation per species. E.g. assessing how good a model deals with an excessive number of zeros per species.

That brings me to my actual question....suppose I want to fit a gllvm of the form:

log(mu) = random effect + Intercepts + Covariates + latent variables 

but inside the 'Covariates` part I have not only covariates but also basis functions of unpenalised smoothers. Like: X1 * b1 + X2 * b2. The X1 and X2 would each have 4 or 5 columns with basis functions. That is not difficult to do. But now I want to do posterior simulation (a la Simon Wood). I want to calculate that X1 * b1  1000 times. For that, I need to have access to the covariance matrix of all the regression parameters. Is that available somewhere so that I can simulate regression parameters?

Posterior simulation would also allow me to do all the nice things that Gavin Simpson is doing in his gratia package.

Kind regards,

Alain

**Comment by BertvanderVeen** [comment]
Hello Alain.

Yes, the residuals due to plot are Dunn-Smyth/randomized quantile residuals as in DHARMa. The covariance matrix of all (fixed) parameters is available either via `se.gllvm(model)` if the model was fitted with `sd.errors = FALSE` and otherwise in `model$Hess$cov.mat.mod`. The covariance matrix of all parameters (fixed and random) is available in `model$Hess$Hess.full`.

Kind regards,
Bert
